src/SFA.DAS.PAS.Account.Api.ClientV2/PasAccountApiClientFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.Extensions.Logging;
+using SFA.DAS.Http;
+using SFA.DAS.PAS.Account.Api.ClientV2.Configuration;
+
+namespace SFA.DAS.PAS.Account.Api.ClientV2
+{
+    internal class PasAccountApiClientFactory
+    {
+        private readonly PasAccountApiConfiguration _configuration;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public PasAccountApiClientFactory(PasAccountApiConfiguration configuration, ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _loggerFactory = loggerFactory;
+        }
+
+        public IPasAccountApiClient CreateClient()
+        {
+            var httpClientFactory = new AzureActiveDirectoryHttpClientFactory(_configuration, _loggerFactory);
+            var httpClient = httpClientFactory.CreateHttpClient();
```

Comment:
The client returned from the factory call implements IDisposable, but this class (which creates it) does not dispose of the instance. There is something off here as the lifetime of the HttpCclient instance is actually significant to the RestHttpClient, so cannot be disposed of until the RestHttpClient is finished with it. As it happens the factory method is only called by IoC to resolve a singleton, so in practice, it will all sort itself out when the app closes down. However, the factory does not know this and if it were to be called in a different way (e.g. the factory were injected into a class and the used to create a client) there would be a problem.

Reply:
Good point